gcp_pilot/healthcare.py

```diff
@@ -175,7 +175,7 @@ def get_page_resources(self) -> Generator[DomainResource]:
             return
 
         for entry in self.response["entry"]:
-            resource_class = get_fhir_model_class(entry["resource"]["resourceType"])
+            resource_class = self.resource_class or get_fhir_model_class(entry["resource"]["resourceType"])
```

Comment:
Do you think we should fail/warn when the `self.resource_class` does not match the response's resourceType?

Also, shouldn't the response's resourceType be more important? I'm thinking about results that might have a bunch of **mixed resource types** (Patient, Appointment, Encounter), and if we use the response's resourceType, we'll be able to properly build those resources.